### PR TITLE
refactor(utilities): improve `codeBlock` and `inlineCodeBlock`

### PR DESCRIPTION
**Pull request by favna**
Now, these functions are more similar to their sister functions in DiscordJS in terms of typing while retaining the additional safety checks that Sapphire has always provided.

### CODE REVIEW

**Review by vladfrangu — DISMISSED** [review]

File: packages/utilities/src/lib/codeBlock.ts [hunk 1]
```diff
@@ -1,14 +1,27 @@
...
+	if (typeof content === 'string') {
+		if (content.length === 0) {
+			return `\`\`\`\n${zws}\n\`\`\``;
+		}
+
+		return `\`\`\`${language}\n${content.replace(/```/, `\`${zws}\`\``).replace(/`$/g, `\`${zws}`)}\n\`\`\``;
 	}
-	return `\`\`\`${language}\n${expression || zws}\`\`\``;
+
+	return typeof content === 'undefined' ? `\`\`\`\n${language}\n\`\`\`` : `\`\`\`${language}\n${content || zws}\n\`\`\``;
```
Comment (kyranet) [inline packages/utilities/src/lib/codeBlock.ts:26]: Is this branch even possible? We already have line 18 for this.

**Review by kyranet — CHANGES_REQUESTED** [review]

File: packages/utilities/src/lib/codeBlock.ts (hunk 1 above)
Comment (favna) [reply packages/utilities/src/lib/codeBlock.ts:26]: If someone in JS writes something like:
```ts
const myCode = someFunc(); // Lets say this returns undefined for whatever reason

codeBlock('js', myCode);
```
then line 18 will be false

So not necessarily

Comment (kyranet) [reply packages/utilities/src/lib/codeBlock.ts:26]: That's literally what line 18 does, it already covers the `string` case.

Comment (favna) [reply packages/utilities/src/lib/codeBlock.ts:26]: `myCode` in my example is undefined so it's NOT covered by the `string` case

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**